### PR TITLE
feat: add Julia support

### PR DESCRIPTION
**Pull request by ShalokShalom**
Hi lovely people 😃 

This contribution is supposed to include Julia into the list of supported languages. 
The highlights.scm file got copied from a repo, that uses the Apache 2 license. 

I think this means, we are supposed to add that one to the project.

### CODE REVIEW

**Review by sammy-ette — CHANGES_REQUESTED** [review]

File: queries/julia/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  (#any-of? @constant.builtin "nothing" "missing"))
+
+(character_literal) @character
+(escape_sequence) @string.escape
+
+(string_literal) @string
+(prefixed_string_literal
+  prefix: (identifier) @function.macro) @string
+
+(command_literal) @string.special
```
[the remark this answers is not in the document]
Comment (ShalokShalom) [reply queries/julia/highlights.scm:179]: Honestly, I dont know much about scm files and treesitter, feel free to adopt as you see fit. 

I would probably just do something stupid. 😄

File: languages.lua [hunk 2]
```diff
@@ -37,6 +38,7 @@ M.exts = {
 	go = 'https://github.com/tree-sitter/tree-sitter-go',
 	gomod = 'https://github.com/camdencheek/tree-sitter-go-mod',
 	javascript = 'https://github.com/tree-sitter/tree-sitter-javascript',
+	julia = 'https://github.com/tree-sitter/tree-sitter-julia'
```
Comment (TorchedSammy) [inline languages.lua:41]: missing comma

File: queries/julia/highlights.scm (hunk 1 above)
[the remark this answers is not in the document]
Comment (TorchedSammy) [reply queries/julia/highlights.scm:179]: just change `@string.special` to `@string`?

File: queries/julia/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  (line_comment)
+  (block_comment)
+] @comment
+
+;;; Punctuation
+
+(quote_expression ":" @symbol)
+["::" "." "," "..."] @punctuation.delimiter
+["[" "]" "(" ")" "{" "}"] @punctuation.bracket
+
```
Comment (TorchedSammy) [inline queries/julia/highlights.scm:193]:
```suggestion

(ERROR) @error
```

**Review by sammy-ette — CHANGES_REQUESTED** [review]
some simple changes, and i'll merge

**Comment by ShalokShalom** [comment]
I think I got everything.

**Review by sammy-ette — APPROVED** [review]

**Comment by eitsupi** [comment]
> The highlights.scm file got copied from a repo, that uses the Apache 2 license.

It seems that this file is under MPL-2.0. See lapce/lapce#2640
Would relicensing be a violation of the license?

**Pull request merged**